addons/account/models/account_bank_statement.py

```diff
@@ -703,7 +703,7 @@ def _prepare_counterpart_move_line_vals(self, counterpart_vals, move_line=None):
         if foreign_currency != company_currency and self.foreign_currency_id:
             amount_currency = amounts[foreign_currency.id]
             currency_id = foreign_currency.id
-        elif journal_currency != company_currency and not self.foreign_currency_id:
+        elif journal_currency != company_currency and (not self.foreign_currency_id or self.foreign_currency_id == company_currency):
```

Comment:
This is completely wrong because all amounts must be somewhere on the journal items. If you set a foreign currency amount, this amount must land on the journal item having the suspense account.